core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -399,16 +401,35 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
                     }
                     DBIO.seq(inserts: _*)
                   } andThen {
-                  DBIO.successful(destWorkspaceContext.workspace)
+                  DBIO.successful((sourceWorkspaceContext, destWorkspaceContext))
                 }
               }
             }
           }
+        }.map { case (sourceWorkspaceContext, destWorkspaceContext) =>
+          //we will fire and forget this. a more involved, but robust, solution involves using the Google Storage Transfer APIs
+          //in most of our use cases, these files should copy quickly enough for there to be no noticeable delay to the user
+          //we also don't want to block returning a response on this call because it's already a slow endpoint
+          destWorkspaceRequest.copyFilesWithPrefix.foreach { prefix =>
+            copyBucketFiles(sourceWorkspaceContext, destWorkspaceContext, prefix)
+          }
+
+          destWorkspaceContext.workspace
         }
       }
     }
   }
 
+  private def validateFileCopyPrefix(copyFilesWithPrefix: String): Unit = {
+    if(copyFilesWithPrefix.equalsIgnoreCase("")) throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, """You may not specify an empty string for `copyFilesWithPrefix`. Did you mean to specify "/" or leave the field out entirely?"""))
+  }
+
+  private def copyBucketFiles(sourceWorkspaceContext: SlickWorkspaceContext, destWorkspaceContext: SlickWorkspaceContext, copyFilesWithPrefix: String) = {
+    gcsDAO.listObjectsWithPrefix(sourceWorkspaceContext.workspace.bucketName, copyFilesWithPrefix).map { objectsToCopy =>
```

Comment:
Should this be a `flatMap`?

Include the return type on this method, and make sure it's a `Future[Something]` not a `Future[Future[Something]]`

Reply:
I know it's currently fire-and-forget so it doesn't REALLY matter, but still good practice to not nest `Future`s